src/block_ffm.rs

```diff
@@ -193,7 +211,13 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
                             } else {
                                 for _z in 0..ffm_fields_count_as_usize {
                                     _mm_prefetch(mem::transmute::<&f32, &i8>(&ffm_weights.get_unchecked(feature_index + ffmk_as_usize)), _MM_HINT_T0);
-                                    for k in 0..ffmk_as_usize {
+                                    for k in (0..ffmk_end).step_by(STEP) {
+                                        let ffm_weights_simd = f32x4::from_slice(ffm_weights.get_unchecked(feature_index + k..feature_index + k + STEP));
+                                        let contra_fields_simd = f32x4::from_slice(contra_fields.get_unchecked(offset + k..offset + k + STEP));
+                                        let result_simd = (feature_value_simd * ffm_weights_simd + contra_fields_simd);
```

Comment:
Something tells me this multiplication is often zero ;) let's address in a separate PR perhaps

Reply:
You mean because of the FMA or because the feature values/weights are 0?

Reply:
values yes